dask/array/slicing.py

```diff
@@ -590,9 +602,29 @@ def take(outname, inname, chunks, index, axis=0):
             PerformanceWarning,
             stacklevel=6,
         )
-
-    index_lists = [idx for _, idx in plan]
-    where_index = [i for i, _ in plan]
+    index = np.asarray(index)
+
+    # Check for chunks from the plan that would violate the user's
+    # configured chunk size.
+    nbytes = utils.parse_bytes(config.get("array.chunk-size"))
+    other_chunks = [chunks[i] for i in range(len(chunks)) if i != axis]
```

Comment:
This code path might never be hit with unknown chunk sizes, but if it is, are unknown chunks properly handled (and if so, could you add a test for this case)?

Reply:
Thanks. It's not handled "correctly" in the sense that we avoid creating a large block, since we don't actually know what the size in bytes is. But it does at least compute, as long as you aren't slicing on an axis with unknown chunk sizes (that case hits https://github.com/dask/dask/issues/6586).